File: smartlist/src/main/java/com/smartlist/services/CryptoService.java
package com.smartlist.services;

import com.smartlist.config.PropertiesConfig;
import io.jsonwebtoken.JwtBuilder;
import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;
import io.jsonwebtoken.io.Decoders;
import io.jsonwebtoken.security.Keys;
import io.jsonwebtoken.security.SecureDigestAlgorithm;
import org.apache.commons.lang3.RandomStringUtils;
import org.apache.commons.lang3.time.DateUtils;
import org.springframework.security.crypto.bcrypt.BCrypt;
import org.springframework.stereotype.Service;

import javax.crypto.SecretKey;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Base64;
import java.util.Date;
import java.util.Map;
import java.util.Random;

@Service
public class CryptoService {

    //Autowired
    final PropertiesConfig propertiesConfig;

    //Global
    private final Random random = new Random();

    public CryptoService(final PropertiesConfig propertiesConfig){
        this.propertiesConfig = propertiesConfig;
    }

    public boolean comparePassword(String plainTextPassword, String hashedPassword){
        return BCrypt.checkpw(plainTextPassword, hashedPassword);
    }

    public String generatePassword(String plainPassword){
        int saltRounds = 10;
        return BCrypt.hashpw(plainPassword, BCrypt.gensalt(saltRounds));

    }

    public String generateToken() {
        return generateToken(50);
    }

    public String generateToken(int size) {
        return RandomStringUtils.randomAlphanumeric(size);
    }

    public String generateHash(String stringToHash) throws NoSuchAlgorithmException {
        MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
        messageDigest.update(stringToHash.getBytes());
        return Base64.getEncoder().encodeToString(messageDigest.digest());

    }

    public String generateRandomString(){
        return RandomStringUtils.random(16, true, true);
    }


}

File: javascript/smartlist/smartlist-api-mongodb-project/src/providers/crypto.service.ts
import { Injectable } from '@nestjs/common';
import * as bcrypt from 'bcrypt';
import * as crypto from 'crypto';

@Injectable()
export class CryptoService {
  private saltRounds = 12;

  async comparePassword(
    toCompare: string,
    comparedTo: string,
  ): Promise<boolean> {
    return bcrypt.compare(toCompare, comparedTo);
  }

  async generatePassword(plainPassword: string): Promise<string> {
    return bcrypt.hash(plainPassword, this.saltRounds);
  }

  generateToken(): string {
    const buffer = crypto.randomBytes(48);
    const token = crypto.createHash('sha1').update(buffer).digest('hex');
    return token;
  }

  generateHash(data: string): string {
    const token = crypto.createHash('sha1').update(data).digest('hex');
    return token;
  }

  generateRandomString(): string {
    return crypto.randomBytes(16).toString('hex');
  }
}
